build/scripts/build.ps1

```diff
@@ -192,18 +190,37 @@ function Make-BootstrapBuild() {
     if ($buildCoreClr) {
         $bootstrapFramework = "netcoreapp2.0"
         $logDir = Join-Path $binariesDir "Logs"
-        Create-Directory $logDir
-        Exec-Console "dotnet" "publish --no-restore src/Compilers/CSharp/csc -o `"$dir/bincore`" --framework $bootstrapFramework $bootstrapArgs -bl:$logDir/BootstrapCsc.binlog"
-        Exec-Console "dotnet" "publish --no-restore src/Compilers/VisualBasic/vbc -o `"$dir/bincore`" --framework $bootstrapFramework $bootstrapArgs -bl:$logDir/BootstrapVbc.binlog"
-        Exec-Console "dotnet" "publish --no-restore src/Compilers/Server/VBCSCompiler -o `"$dir/bincore`" --framework $bootstrapFramework $bootstrapArgs -bl:$logDir/BootstrapVBCSCompiler.binlog"
-        Exec-Console "dotnet" "publish --no-restore src/Compilers/Core/MSBuildTask -o `"$dir`" --framework $bootstrapFramework $bootstrapArgs -bl:$binariesDir/BootstrapMSBuildTask.binlog"
+        $projectFiles = @(
+            'src/Compilers/CSharp/csc/csc.csproj',
+            'src/Compilers/VisualBasic/vbc/vbc.csproj',
+            'src/Compilers/Server/VBCSCompiler/VBCSCompiler.csproj',
+            'src/Compilers/Core/MSBuildTask/MSBuildTask.csproj'
+        )
+
+        foreach ($projectFilePath in $projectFiles) { 
+            $fileName = [IO.Path]::GetFileNameWithoutExtension((Split-Path -leaf $projectFilePath))
+            $logFileName = "Bootstrap$($fileName)"
+            Run-MSBuild $projectFilePath "/t:Publish /p:TargetFramework=netcoreapp2.0 $bootstrapArgs" -logFileName $logFileName -useDotnetBuild
+        }
+
+        Run-MSBuild "src/Interactive/csi/csi.csproj" -logFileName "BootstrapCsi" -useDotnetBuild
+
+        Ensure-NuGet | Out-Null
+        Exec-Console "$configDir\Exes\csi\net46\csi.exe" "$repoDir\src\NuGet\BuildNuGets.csx $configDir 1.0.0-bootstrap $dir `"<developer build>`" Microsoft.NETCore.Compilers.nuspec"
+        Unzip-File "$dir\Microsoft.NETCore.Compilers.1.0.0-bootstrap.nupkg" "$dir\Microsoft.NETCore.Compilers\1.0.0"
+
+        Write-Host "Cleaning Bootstrap compiler artifacts"
+        Run-MSBuild "Compilers.sln" "/t:Clean"
         Stop-BuildProcesses
     }
     else {
         Run-MSBuild "build\Toolset\Toolset.csproj" $bootstrapArgs -logFileName "Bootstrap"
         Remove-Item -re $dir -ErrorAction SilentlyContinue
         Create-Directory $dir
-        Move-Item "$configDir\Exes\Toolset\*" $dir
+
+        Ensure-NuGet | Out-Null
+        Exec-Console "$configDir\Exes\csi\net46\csi.exe" "$repoDir\src\NuGet\BuildNuGets.csx $configDir 1.0.0-bootstrap $dir `"<developer build>`" Microsoft.Net.Compilers.nuspec"
```

Comment:
In the case above, we build `csi.csproj` before using `csi.exe`. Should we do the same here?

Reply:
This isn't necessary here because it's included inside Toolset.csproj hence it's already built. 

I can see how this is confusing ... mostly because it's confusing :frown:. 


---
In reply to: [174913794](https://github.com/dotnet/roslyn/pull/25440#discussion_r174913794) [](ancestors = 174913794)